p2panda-rs/src/entry/entry_signed.rs

```diff
@@ -191,4 +362,40 @@ mod tests {
         let key_value_retrieved = hash_map.get(&entry_first_encoded).unwrap().to_owned();
         assert_eq!(key_value, key_value_retrieved)
     }
+
+    proptest! {
+        #[test]
+        fn non_standard_strings_dont_crash(ref s in "\\PC*") {
+            let result = EntrySigned::new(s);
+
+            assert!(result.is_err())
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn partially_correct_strings_dont_crash(
+            ref author in "[0-9a-f]{64}|[*]{0}",
+            ref log_id in prop::num::u64::ANY,
+            ref seq_num in prop::num::u64::ANY,
+            ref skiplink in "[0-9a-f]{68}|[*]{0}",
+            ref backlink in "[0-9a-f]{68}|[*]{0}",
+            ref payload_size in prop::num::u64::ANY,
+            ref payload_hash in "[0-9a-f]{68}|[*]{0}",
+            ref signature in "[0-9a-f]{68}|[*]{0}"
+        ) {
+            let encoded_entry = "0".to_string()
+                + author
+                + log_id.to_string().as_str()
+                + seq_num.to_string().as_str()
+                + skiplink
+                + backlink
+                + payload_size.to_string().as_str()
+                + payload_hash
+                + signature;
+            let result = EntrySigned::new(&encoded_entry);
+
+            assert!(result.is_err())
```

Comment:
I thought of something else with prop testing: If these eventually crash due to some randomly generated value it might be very hard to recreate that value if we don't have a record of it. Maybe all prop tests can include a logging call when they error?

Reply:
(in this case - when they don't error?)

Reply:
No wait ... in this case when this panics, which we can't predict ... so probably everytime.